src/sqlfluff/api/simple.py

```diff
@@ -1,20 +1,23 @@
 """The simple public API methods."""
 
+from typing import Any, Dict, List, Optional, TextIO, Union
 from sqlfluff.core import Linter
+from sqlfluff.core import SQLBaseError
+from sqlfluff.core.linter import ParsedString
 
 
 class APIParsingError(ValueError):
     """An exception which holds a set of violations."""
 
-    def __init__(self, violations, **kwargs):
+    def __init__(self, violations: List[SQLBaseError], *args: Any):
         self.violations = violations
         self.msg = f"Found {len(violations)} issues while parsing string."
         for viol in violations:
             self.msg += f"\n{viol!s}"
-        super().__init__(self.msg, **kwargs)
+        super().__init__(self.msg, *args)
 
 
-def _unify_str_or_file(sql):
+def _unify_str_or_file(sql: Union[str, TextIO]) -> str:
```

Comment:
I would like to propose removing this from the simple API.

The simple api should just take a str, rather than also taking file objects. I think it is better defined to say that `fix` is a function that takes a string and returns a string, and then the user can get that string from wherever they need.
i.e.
```
with open("<path_to_sql_file>", "r") as f:
    sql = f.read()

fixed_sql = sqlfluff.fix(sql)
```

Reply:
This sounds good to me, and the type hinting work helps highlight this -- is the added complexity in the API worthwhile? Probably not.

Reply:
Agreed one of my favourite things about type hinting is how it highlights unnecessary complexity/inconsistencies. I'll make this change!